crates/rattler_installs_packages/src/python_env/venv.rs

```diff
@@ -122,31 +136,164 @@ impl VEnv {
+
+        let exe_path = install_paths.scripts().join(base_python_name);
+        let abs_exe_path = venv_abs_dir.join(exe_path);
+
+        Self::setup_python(&abs_exe_path, &base_python_path, base_python_version)?;
+
+        Ok(VEnv::new(venv_abs_dir.to_path_buf(), install_paths))
+    }
+
+    /// Create all directories based on venv install paths mapping
+    pub fn create_install_paths(
+        venv_abs_path: &Path,
+        install_paths: &InstallPaths,
+    ) -> std::io::Result<()> {
+        if !venv_abs_path.exists() {
+            fs::create_dir_all(venv_abs_path)?;
+        }
+
+        let libpath = Path::new(&venv_abs_path).join(install_paths.site_packages());
+        let include_path = Path::new(&venv_abs_path).join(install_paths.include());
+        let bin_path = Path::new(&venv_abs_path).join(install_paths.scripts());
+
+        let paths_to_create = [libpath, include_path, bin_path];
+
+        for path in paths_to_create.iter() {
+            if !path.exists() {
+                fs::create_dir_all(path)?;
+            }
+        }
+
+        // https://bugs.python.org/issue21197
+        // create lib64 as a symlink to lib on 64-bit non-OS X POSIX
+        #[cfg(all(target_pointer_width = "64", unix, not(target_os = "macos")))]
+        {
+            let lib64 = venv_abs_path.join("lib64");
+            if !lib64.exists() {
+                std::os::unix::fs::symlink("lib", lib64)?;
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Create pyvenv.cfg and write it's content based on system python
+    pub fn create_pyvenv(
+        venv_path: &Path,
+        python_path: &Path,
+        python_version: PythonInterpreterVersion,
+    ) -> std::io::Result<()> {
+        let venv_name = venv_path
+            .file_name()
+            .and_then(OsStr::to_str)
+            .ok_or_else(|| {
+                std::io::Error::new(
+                    ErrorKind::InvalidData,
+                    format!(
+                        "cannot extract base name from venv path {}",
+                        venv_path.display()
+                    ),
+                )
+            })?;
+
+        let pyenv_cfg_content = format!(
+            r#"
+home = {}
+include-system-site-packages = false
+version = {}.{}.{}
+prompt = {}"#,
+            python_path
+                .parent()
+                .expect("system python path should have parent folder")
+                .display(),
+            python_version.major,
+            python_version.minor,
+            python_version.patch,
+            venv_name,
+        );
+
+        let cfg_path = Path::new(&venv_path).join("pyvenv.cfg");
+        std::fs::write(cfg_path, pyenv_cfg_content)?;
+        Ok(())
+    }
+
+    /// Copy original python executable and populate other suffixed binaries
+    pub fn setup_python(
+        venv_exe_path: &Path,
+        original_python_exe: &Path,
+        python_version: PythonInterpreterVersion,
```

Comment:
Don't know if we can disable unused in this case with a `cfg_attr`?

Reply:
Thanks a lot for this idiomatic suggestion